MapVisualisationExample-WW2Cargo/IntMaps.ipynb

```diff
@@ -0,0 +1,305 @@
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2fe40785-eab5-4392-ab5a-2615d6ced8ac",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#df.rename(index={2: \"GRT\"})\n",
+    "ax = df[2].plot.hist(bins=30, range=[0, 16000])\n",
+    "plt.axvline(x=7174.0, color=\"orange\")\n",
+    "#plt.axvline(x=4963.0, color=\"orange\")\n",
+    "ax.set_xlabel(\"Cargo (GRT)\")\n",
+    "ax.set_ylabel(\"Ships Lost\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2f6a4451-e8a1-4d21-a9b8-edf51c1de3c0",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "pio.renderers.default = 'iframe'\n",
+    "\n",
+    "# Setup a colour index by year\n",
+    "colourindex = dict(zip(map(str, range(1939,1946)), \n",
+    "                      [\"lightblue\",\"yellow\",\"orange\",\"red\",\"violet\",\"darkgrey\", \"grey\"]))\n",
+    "\n",
+    "fig1 = px.scatter_mapbox(lat=df[9], lon=df[10], hover_name=df[1], hover_data={\"Sunk\": df[0], \"Nationality\":df[3], \"Tonnage\":df[2]}, \n",
+    "                        color_discrete_sequence=px.colors.qualitative.Alphabet,color=df['year'],\n",
+    "                        color_discrete_map=colourindex, animation_frame=df[\"year\"],\n",
+    "                        zoom=3, labels={\"color\": \"Year\"})\n",
+    "fig1.update_layout(\n",
+    "    mapbox_style=\"white-bg\",\n",
+    "    mapbox_layers=[\n",
+    "        {\n",
+    "            \"below\": 'traces',\n",
+    "            \"sourcetype\": \"raster\",\n",
+    "            \"sourceattribution\": \"United States Geological Survey\",\n",
+    "            \"source\": [\n",
+    "                \"https://basemap.nationalmap.gov/arcgis/rest/services/USGSImageryOnly/MapServer/tile/{z}/{y}/{x}\"\n",
+    "            ]\n",
+    "        }\n",
+    "      ])\n",
+    "fig1.update_layout(margin={\"r\":0,\"t\":40,\"l\":0,\"b\":0})\n",
+    "\n",
+    "#fig1.show()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3e260177-742f-4857-a13d-dd2f07b4f55b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def openurl(url):\n",
+    "    IPython.display.display(IPython.display.Javascript('window.open(\"{url}\");'.format(url=url)))\n",
+    "\n",
+    "figw = go.FigureWidget(layout={'hovermode': 'closest'})\n",
+    "mapi = figw.add_scattergeo( lat = df[9],lon = df[10], mode=\"markers\", text=df[1], marker=dict(color=\"red\", opacity=0.4) )\n",
+    "                       \n",
+    "\n",
+    "mapi.update_layout(title = 'Test links')\n",
+    "mapi.update_layout(\n",
+    "    title = '<span style=\"font-size: 14px;\">Ships Lost to U-Boat Campaign (\\'39-\\'45)</span>' +\n",
+    "               '<i><span style=\"font-size: 12px;\">   Data are clickable</span></i>',\n",
+    "    mapbox_style=\"white-bg\",\n",
+    "    mapbox_layers=[ # Why is this not working properly !?! & why is the map frame itself zooming out !\n",
```

Comment:
Should the comment in this line be removed?

Reply:
Definitely  ! I've cleaned up the comments a bit & removed the commented-out lines. Also put in a quick check for the data location.